vision/beta/cloud-client/src/main/java/com/example/vision/DetectBatchAnnotateFiles.java

```diff
@@ -0,0 +1,118 @@
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.vision.v1p4beta1.AnnotateFileRequest;
+import com.google.cloud.vision.v1p4beta1.AnnotateFileResponse;
+import com.google.cloud.vision.v1p4beta1.BatchAnnotateFilesRequest;
+import com.google.cloud.vision.v1p4beta1.BatchAnnotateFilesResponse;
+import com.google.cloud.vision.v1p4beta1.Block;
+import com.google.cloud.vision.v1p4beta1.Feature;
+import com.google.cloud.vision.v1p4beta1.Feature.Type;
+import com.google.cloud.vision.v1p4beta1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1p4beta1.InputConfig;
+import com.google.cloud.vision.v1p4beta1.Page;
+import com.google.cloud.vision.v1p4beta1.Paragraph;
+import com.google.cloud.vision.v1p4beta1.Symbol;
+import com.google.cloud.vision.v1p4beta1.TextAnnotation;
+import com.google.cloud.vision.v1p4beta1.Word;
+import com.google.protobuf.ByteString;
+
+import java.io.FileInputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+
+public class DetectBatchAnnotateFiles {
+
+  /**
+   * Performs document feature detection on a local PDF/TIFF/GIF file.
+   * While your PDF file may have several pages, this API can process up to 5 pages only.
+   * @param filePath The local file path e.g. "path/to/file.pdf"
+   * @throws Exception on errors
+   */
+  public static void detectBatchAnnotateFiles(String filePath) {
+
+    // Initialize the client that will be used to send requests. This client only needs to be
+    // created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (ImageAnnotatorClient client = ImageAnnotatorClient.create()) {
+      //Annotate the first two pages and the last one (max 5 pages)
+      //First page starts at 1, and not 0. Last page is -1.
+      List<Integer> pages = Arrays.asList(1,2,-1);
+      ByteString pdfBytes = ByteString.readFrom(new FileInputStream(filePath));
+      Feature feat = Feature.newBuilder().setType(Type.DOCUMENT_TEXT_DETECTION).build();
+      //Other supported mime types : 'image/tiff' or 'image/gif'
+      InputConfig inputConfig =
+          InputConfig.newBuilder().setMimeType("application/pdf").setContent(pdfBytes).build();
+      AnnotateFileRequest request =
+          AnnotateFileRequest.newBuilder()
+              .addFeatures(feat)
+              .setInputConfig(inputConfig)
+              .addAllPages(pages)
+              .build();
+      List<AnnotateFileRequest> requests = new ArrayList<>();
+      requests.add(request);
+      BatchAnnotateFilesRequest batchAnnotateFilesRequest =
+          BatchAnnotateFilesRequest.newBuilder().addAllRequests(requests).build();
+      ApiFuture<BatchAnnotateFilesResponse> future =
+          client.batchAnnotateFilesCallable().futureCall(batchAnnotateFilesRequest);
+      BatchAnnotateFilesResponse response = future.get();
+      client.close();
```

Comment:
Since the `client` is in a try, it should be closed automatically.

Reply:
Fixed in latest commit.